backend/src/main/java/asegroup1/api/daos/DaoImpl.java

```diff
@@ -0,0 +1,66 @@
+package asegroup1.api.daos;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceContextType;
+import java.util.List;
+
+/**
+ * @author Greg Mitten
+ * gregoryamitten@gmail.com
+ */
+
+public class DaoImpl<T> implements Dao<T> {
+
+    /**
+     * A necessary parameter in many of the entityManager's methods
+     */
+    private Class currentClass;
```

Comment:
Provide Class with the generic parameter T, i.e. Class<T>. This will remove the unchecked type warnings in this file.